src/dataflow/src/render/upsert.rs

```diff
@@ -150,36 +207,65 @@ where
                                             Ok(value) => {
                                                 if let Some(value) = value {
                                                     // prepend key to row
```

Comment:
Merge this comment with the "unpack rows" comments below to make it clear that it's all happening at the same time?